cmd/snap/cmd_run.go

```diff
@@ -307,6 +319,21 @@ func (x *cmdRun) snapRunHook(snapName string) error {
 	return x.runSnapConfine(info, hook.SecurityTag(), snapName, hook.Name, nil)
 }
 
+func (x *cmdRun) snapRunTimer(snapApp, timer string, args []string) error {
+	schedule, err := timeutil.ParseSchedule(timer)
+	if err != nil {
+		return fmt.Errorf("invalid timer format: %v", err)
+	}
+
+	now := timeNow()
+	if !timeutil.Includes(schedule, now) {
+		fmt.Fprintf(Stderr, "on %v attempted to run %v timer outside of scheduled time %q\n", now, snapApp, timer)
```

Comment:
The start of the "on %v attempted to run %v timer outside of scheduled time %q\n" sentence sounds a bit weird, perhaps worth checking if the order is ok with a native english speaker.

Reply:
@chipaca how does that sound to you?

Reply:
example output:
```
on 2018-02-21 16:16:32.540206021 +0100 CET m=+0.003731543 attempted to run hello-world.env timer outside of scheduled time "10:00-12:00"
```

Reply:
Give you're printing straight to stderr, I'd suggest
```go
fmt.Fprintf(Stderr, "%s: attempted to run %q timer outside of scheduled time %q\n", now.Format(time.RFC3339), snapApp, timer)
```

Reply:
although it's unclear to me how relevant that time is, at least to that level of detail.

It'd probably be better to do something like
```
cannot run %q timer: the scheduled time (%q) ended %s ago, and does not happen again for another %s 
```

Reply:
Thanks for the suggestions @chipaca . I'll go with your first proposal. The second option may require some changes in `timeutil` package and I'm not that keen to introduce more changes there at this time.